src/app/application.cpp

```diff
@@ -645,41 +647,53 @@ int Application::exec(const QStringList &params)
     Net::DownloadManager::initInstance();
     IconProvider::initInstance();
 
-    try
-    {
-        BitTorrent::Session::initInstance();
-        connect(BitTorrent::Session::instance(), &BitTorrent::Session::restored, this, [this]()
-        {
-#ifndef DISABLE_WEBUI
-            m_webui = new WebUI(this);
-#ifdef DISABLE_GUI
-            if (m_webui->isErrored())
-                QCoreApplication::exit(1);
-            connect(m_webui, &WebUI::fatalError, this, []() { QCoreApplication::exit(1); });
+    BitTorrent::Session::initInstance();
+#ifndef DISABLE_GUI
+    UIThemeManager::initInstance();
 
-            const Preferences *pref = Preferences::instance();
+    m_desktopIntegration = new DesktopIntegration(this);
+    m_desktopIntegration->setToolTip(tr("Loading torrents..."));
+#ifndef Q_OS_MACOS
```

Comment:
Why this?

Reply:
> Why this?

I just copied the existing logic from the `MainWindow`. It doesn't provide `Exit` action in the case of macOS (don't force me to figure out why it's done this way!), and since there are no more actions here, we don't need to create the menu itself, do we?